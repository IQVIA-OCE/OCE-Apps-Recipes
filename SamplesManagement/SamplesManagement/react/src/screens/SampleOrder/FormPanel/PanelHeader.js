import React from 'react';
import { View, StyleSheet } from 'react-native';
import FormDetails from '../../../components/FormDetails/FormDetails';
import Status from '../../../components/Status/Status';
import { Paragraph, Checkbox, themeGrey, useTheme } from 'apollo-react-native';
import { useFormikContext } from 'formik';

const PanelHeader = ({ readonly }) => {
  const context = useFormikContext();
  const theme = useTheme();

  const { values, setFieldValue } = context;

  return (
    <View style={[styles.container, { backgroundColor: theme.colors.surface }]} testID="PanelHeader">
      <FormDetails title="Order Recipient">
        <Paragraph style={{ fontSize: 16 }}>
          {values.fields.user ? values.fields.user.Name : ''}
        </Paragraph>
      </FormDetails>
      <FormDetails title="Recipient Territory">
        <Paragraph style={{ fontSize: 16 }}>
          {values.fields.territory ? values.fields.territory.name : ''}
        </Paragraph>
      </FormDetails>
      <FormDetails title="Status">
        <Status size="large" status={values.fields.status} />
      </FormDetails>
      <FormDetails>
        <Checkbox
          onPress={() => setFieldValue('fields.isUrgent', !values.fields.isUrgent)}
          status={values.fields.isUrgent == true ? 'checked' : null}
          label="Is Urgent?"
          disabled={readonly}
        />
      </FormDetails>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: themeGrey[200],
  },
});

export default PanelHeader;
