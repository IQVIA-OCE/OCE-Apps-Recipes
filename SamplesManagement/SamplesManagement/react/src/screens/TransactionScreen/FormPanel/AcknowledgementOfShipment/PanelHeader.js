import React from 'react';
import { View, StyleSheet } from 'react-native';
import FormDetails from '../../../../components/FormDetails/FormDetails';
import moment from 'moment';
import Status from '../../../../components/Status/Status';
import { Colors, Paragraph } from 'apollo-react-native';
import { useFormikContext } from 'formik';

const PanelHeader = () => {
  const context = useFormikContext();
  const { values } = context;

  return (
    <View style={styles.container}>
      <FormDetails title="Transaction Date Time">
        <Paragraph style={{ fontSize: 16 }}>
          {moment(values.fields.transactionDateTime).format(
            'MMM D YYYY hh:mm a'
          )}
        </Paragraph>
      </FormDetails>
      <FormDetails title="Status">
        <Status size="large" status={values.fields.status} />
      </FormDetails>
      <FormDetails title="Transaction Rep">
        <Paragraph style={{ fontSize: 16 }}>{values.fields.transactionRep ? values.fields.transactionRep.Name : ''}</Paragraph>
      </FormDetails>
      <FormDetails title="Transaction Rep Territory">
        <Paragraph style={{ fontSize: 16 }}>{values.fields.territory.name}</Paragraph>
      </FormDetails>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: Colors.themeGrey[200],
  },
});

export default PanelHeader;
