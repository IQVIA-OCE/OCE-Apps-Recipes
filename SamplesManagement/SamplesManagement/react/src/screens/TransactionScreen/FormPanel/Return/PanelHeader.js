import React from 'react';
import { View, StyleSheet } from 'react-native';
import FormDetails from '../../../../components/FormDetails/FormDetails';
import moment from 'moment';
import Status from '../../../../components/Status/Status';
import { Paragraph, themeGrey } from 'apollo-react-native';
import { environment } from '../../../../../bridge/EnvironmentData/EnvironmentData.native';

const PanelHeader = () => {
  const territory = environment.territory();
  return (
    <View style={styles.container}>
      <FormDetails title="Transaction Date Time">
        <Paragraph style={{ fontSize: 16 }}>
          {moment().format('MMM D YYYY hh:mm a')}
        </Paragraph>
      </FormDetails>
      <FormDetails title="Status">
        <Status size="large" status={'In Progress'} />
      </FormDetails>
      <FormDetails title="Transaction Rep">
        <Paragraph style={{ fontSize: 16 }}>Salesrep</Paragraph>
      </FormDetails>
      <FormDetails title="Transaction Rep Territory">
        <Paragraph style={{ fontSize: 16 }}>{territory.name}</Paragraph>
      </FormDetails>
      <FormDetails title="Ship to" />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: themeGrey[200],
  },
});

export default PanelHeader;
