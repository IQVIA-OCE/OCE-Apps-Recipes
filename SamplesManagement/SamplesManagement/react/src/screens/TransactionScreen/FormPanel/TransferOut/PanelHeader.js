import React from 'react';
import { View, StyleSheet } from 'react-native';
import FormDetails from '../../../../components/FormDetails/FormDetails';
import moment from 'moment';
import Status from '../../../../components/Status/Status';
import { Paragraph, themeGrey, useTheme } from 'apollo-react-native';
import { useFormikContext } from 'formik';

const PanelHeader = () => {
  const context = useFormikContext();
  const { values } = context;
  const theme = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: theme.colors.surface }]} testID="TransferOutPanelHeader">
      <FormDetails title="Transaction Date Time">
        <Paragraph style={{ fontSize: 16 }}>
          {moment(values.fields.transactionDateTime).format(
            'MMM D YYYY hh:mm a'
          )}
        </Paragraph>
      </FormDetails>
      <FormDetails title="Status">
        <Status size="large" status={values.fields.status} />
      </FormDetails>
      <FormDetails title="From Sales Rep">
        <Paragraph style={{ fontSize: 16 }}>{values.fields.user ? values.fields.user.Name : ''}</Paragraph>
      </FormDetails>
      <FormDetails title="From Sales Rep Territory">
        <Paragraph style={{ fontSize: 16 }}>{values.fields.fromSalesRepTerritory ? values.fields.fromSalesRepTerritory.name : ''}</Paragraph>
      </FormDetails>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: themeGrey[200],
  },
});

export default PanelHeader;
