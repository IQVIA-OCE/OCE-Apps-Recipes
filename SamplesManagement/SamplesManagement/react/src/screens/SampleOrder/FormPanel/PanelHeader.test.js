import React from 'react';
import PanelHeader from './PanelHeader';
import { Formik } from 'formik';
import { render } from '@testing-library/react-native';

describe('PanelHeader', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render properly', () => {
    const { getByTestId } = render(
      <Formik
        initialValues={{
          fields: {
            status: 'In Progress',
            user: { Name: 'name' },
            territory: { name: 'name' },
            isUrgent: false
          },
        }}
      >
        <PanelHeader />
      </Formik>
    )

    expect(getByTestId('PanelHeader')).toBeTruthy();
  });
});
