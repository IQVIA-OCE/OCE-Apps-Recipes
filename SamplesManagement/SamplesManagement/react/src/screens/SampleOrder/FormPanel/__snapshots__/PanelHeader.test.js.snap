// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader should render properly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#DADADA",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4D4D4D",
          "marginBottom": 5,
        }
      }
    >
      Order Recipient
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16,
              },
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16,
              },
            ],
          ],
        ]
      }
    >
      name
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4D4D4D",
          "marginBottom": 5,
        }
      }
    >
      Recipient Territory
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16,
              },
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16,
              },
            ],
          ],
        ]
      }
    >
      name
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4D4D4D",
          "marginBottom": 5,
        }
      }
    >
      Status
    </Text>
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "borderRadius": 10,
            "flexDirection": "row",
            "paddingHorizontal": 6,
            "paddingVertical": 2,
          },
          Object {
            "borderRadius": 15,
            "fontSize": 16,
            "paddingHorizontal": 8,
            "paddingVertical": 6,
          },
          Object {
            "backgroundColor": "#ffb75d",
            "marginRight": 5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        In Progress
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4D4D4D",
          "marginBottom": 5,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "padding": 6,
                "width": 36,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï„±
            </Text>
            <View
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderWidth": 0,
                    "height": 14,
                    "width": 14,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontWeight": "normal",
            "opacity": 1,
          }
        }
      >
        Is Urgent?
      </Text>
    </View>
  </View>
</View>
`;
